### PR TITLE
Fix a memory leak in TF1::Copy

### PR DESCRIPTION
**Pull request by lmoneta**
Fixes some leaks in TF1::Copy by using unique_ptr as data members 

Do a similar fix also in TFormula

### CODE REVIEW

**Comment by Axel-Naumann** [comment]
Instead of exposing `TF1FunctorPointer` and `TMethodCall` you can also outline all constructors and the destructor of `TF1`. Given that construction and destructions isn't a hot call that would be my preferred solution - especially as it also avoids having to expose `TF1FunctorPointer` to the dictionary. Did / can you try that?

**Comment by Axel-Naumann** [comment]
O wait - I think I understand that there is *also* a different issue: `unique_ptr` cannot delete an object of a non-public type. In that case forget my comment - including TMethodCall is not that relevant. But why do we need to expose `TF1FunctorPointer` to the dictionary?

File: hist/hist/src/TF1.cxx [hunk 1]
```diff
@@ -3619,10 +3603,10 @@ void TF1::Streamer(TBuffer &b)
...
          Save(fXmin, fXmax, 0, 0, 0, 0);
       }
-      if (fType == EFType::kCompositionFcn)
-         fComposition_ptr = fComposition.get();
-      else
-         fComposition_ptr = nullptr;
+      // if (fType == EFType::kCompositionFcn)
+      //    fComposition_ptr = fComposition.get();
+      // else
+      //    fComposition_ptr = nullptr;
```
Comment (Axel-Naumann) [inline hist/hist/src/TF1.cxx:3609]: Remove?
```suggestion
```

**Review by Axel-Naumann — APPROVED** [review]

**Comment by lmoneta** [comment]
Hi, 

 I think there were some issues if TF1FunctorPointer was not in the dictionary, although I don't fully understand why, since it is not a transient data member

**Comment by Axel-Naumann** [comment]
Yeah *because* it's now persistent we want a dictionary for it.

**Comment by lmoneta** [comment]
std::unique_ptr<TF1FunctorPointer>  is a transient member of TF1, I will try to remove it from the dictionary

**Pull request merged**